### DISCUSSION TITLE
swoole 6 - when do you guys plan on releasing it?

### DISCUSSION THREAD

**Discussion by linuxd3v** [swoole/swoole-src#5613] [Q&A] [answered]
swoole 6 - when do you guys plan on releasing it?    
    
Is there anywhere one could see list of changes and improvements?

**Comment by NathanFreeman** [comment, marked as answer]
Swoole 6 will be released soon.

Here are some new features.
- Added multi-threading support, require the ZTS version of PHP. Add `--enable-swoole-thread` option to the configure command to activate it.
- Added a new thread class `Swoole\Thread`. 
- Introduced thread lock `Swoole\Thread\Lock`.
- Added thread atomic counter `Swoole\Thread\Atomic`, `Swoole\Thread\Atomic\Long`. 
- Added safe concurrent containers `Swoole\Thread\Map`, `Swoole\Thread\ArrayList`, `Swoole\Thread\Queue`. 
- File asynchronous operations support `iouring` as the underlying engine. Installing `liburing` and compiling `Swoole` with `--enable-iouring` enables asynchronous operations for functions like `file_get_contents`, `file_put_contents`, `fopen`, `fclose`, `fread`, `fwrite`, `mkdir`, `unlink`, `fsync`, `fdatasync`, `rename`, `fstat`, `lstat`, `filesize` through `iouring`. 
- Upgraded `Boost Context` to version 1.84. Now, Loongson CPUs can also support coroutines. 
- Added `Swoole\Thread\Map::find()` method. 
- Added `Swoole\Thread\ArrayList::find()` method. 
- Added `Swoole\Thread\ArrayList::offsetUnset()` method. 
- Added `Swoole\Process::getAffinity()` method. 
- Added `Swoole\Thread::setName()` method. 
- Added `Swoole\Thread::setAffinity()` method. 
- Added `Swoole\Thread::getAffinity()` method. 
- Added `Swoole\Thread::setPriority()` method. 
- Added `Swoole\Thread::getPriority()` method. 
- Added `Swoole\Thread::gettid()` method.
- The file asynchronous engine `iouring` supports multi-threaded polling mode `IORING_SETUP_SQPOLL`. 
- Added `iouring_workers` to modify the number of `iouring` threads. 
- Added `iouring_flags` to support modifying the `iouring` working mode. 
- Added `Swoole\Thread\Barrier` for multi-thread synchronization barrier. 
- Added new function and class to set cookies. 
- Added `non-blocking, reentrant coroutine mutex lock`, which can be used between processes/threads without blocking them. 
- `Swoole\Coroutine\Socket::getOption()` supports the `TCP_INFO` option. 
- `Swoole\Client` synchronous blocking client supports `http` proxy. 
- Added asynchronous non-blocking `TCP/UDP/Unix socket` client `Swoole\Async\Client`. 
- Optimized the `Swoole\Redis\Server::format()` method to support zero-copy memory, support `redis` nested structure. 
- Supports the high-performance compression tool `Zstd`. You only need to add `--enable-zstd` when compiling `Swoole`, and then `zstd` can be used to compress or decode responses between the `http` client and server.
